src/js/actions/tools.js

```diff
@@ -727,6 +745,43 @@ define(function (require, exports) {
     enterPathModalState.modal = true;
 
     /**
+     * recreate pointer policies if we're in Vector Mask Mode on UI Store change events
+     *
+     * return {Promise}
+     */
+    var handleVectorMaskUIChange = function () {
+        var toolStore = this.flux.store("tool"),
+            vectorMaskMode = toolStore.getVectorMode();
+
+        if (vectorMaskMode) {
+            var pointerPolicyID = toolStore.getVectorMaskPolicyID();
+
+            if (pointerPolicyID) {
+                return this.transfer(policy.removePointerPolicies, pointerPolicyID)
+                    .bind(this)
+                    .then(function () {
+                        var area = _getVectorMaskPolicyBounds.call(this),
+                            pointerPolicy = new PointerEventPolicy(UI.policyAction.ALWAYS_PROPAGATE,
+                                OS.eventKind.LEFT_MOUSE_DOWN,
+                                {},
+                                area);
+                    
+                        return this.transfer(policy.addPointerPolicies, [pointerPolicy]);
+                    })
+                    .then(function (policyID) {
+                        this.dispatch(events.tool.VECTOR_MASK_POLICY_CHANGE, policyID);
```

Comment:
I think this is writing tool store state, so actually you need the `JS_TOOL` write lock.